src/components/Grid.tsx

```diff
@@ -0,0 +1,521 @@
+import React from "react";
+import { CardItem, CardType, Config, SchemeIni, Snippet, TabItemConfig, TabType } from "../types/marketplace-types";
+import { getLocalStorageDataFromKey, generateSchemesOptions, injectColourScheme } from "../logic/Utils";
+import { LOCALSTORAGE_KEYS, ITEMS_PER_REQUEST } from "../constants";
+import { openModal } from "../logic/LaunchModals";
+import {
+  getExtensionRepos, fetchExtensionManifest,
+  getThemeRepos, fetchThemeManifest,
+  fetchCssSnippets, getBlacklist,
+} from "../logic/FetchRemotes";
+import LoadMoreIcon from "./Icons/LoadMoreIcon";
+import LoadingIcon from "./Icons/LoadingIcon";
+import SettingsIcon from "./Icons/SettingsIcon";
+import SortBox from "./Sortbox";
+import { TopBarContent } from "./TabBar";
+import Card from "./Card/Card";
+import Button from "./Button";
+
+export default class Grid extends React.Component<
+{
+  title: string,
+  CONFIG: Config,
+  updateAppConfig: (CONFIG: Config) => void,
+},
+{
+  // TODO: add types
+  searchValue: string,
+  cards: Card[],
+  tabs: TabItemConfig[],
+  rest: boolean,
+  endOfList: boolean,
+  activeThemeKey?: string,
+  schemes: SchemeIni,
+  activeScheme?: string | null,
+}
+> {
+  constructor(props) {
+    super(props);
+    Object.assign(this, props);
+    this.updateAppConfig = props.updateAppConfig.bind(this);
+
+    // Fetches the sorting options, fetched from SortBox.js
+    this.sortConfig = {
+      by: getLocalStorageDataFromKey(LOCALSTORAGE_KEYS.sortBy, "top"),
+    };
+
+    this.state = {
+      searchValue: "",
+      cards: [],
+      tabs: props.CONFIG.tabs,
+      rest: true,
+      endOfList: false,
+      schemes: props.CONFIG.theme.schemes,
+      activeScheme: props.CONFIG.theme.activeScheme,
+      activeThemeKey: props.CONFIG.theme.activeThemeKey,
+    };
+  }
+
+  searchRequested: boolean;
+  endOfList = false;
+  lastScroll = 0;
+  requestQueue: any[] = [];
```

Comment:
```suggestion
  requestQueue: never[][] = [];
```
Not important but if you want to get rid of as most implicit `any` as possible